### PR TITLE
Fix rebuy false appearances

### LINKED ISSUE #681: was asked to rebuy right after I sat at the table
Production
Table https://dapp.acebusters.com/table/0x81b94ee5dbe3c45aab492069928af651a83d052e
Hand 69

One guy was in sit out at the table. I joined with 200 NTZ stack and immediatelly was asked either to rebuy or leave

### LINKED ISSUE #819: I was asked to rebuy everytime I went all-in
Happend every time 

bildschirmfoto 2017-11-27 um 12 08 14

### PR DESCRIPTION
**Pull request by sunify**
Resolves #819 
Resolves #681

Also, now dialog show when player's balance < bb instead of <= 0. We have that logic on backend and player was unable to continue the game because we didn't show rebuy dialog for him.

### CODE REVIEW

File: app/containers/Table/index.js [hunk 1]
```diff
@@ -88,6 +89,39 @@ const SpinnerWrapper = styled.div`
...
+      !standingUp &&
+      !rebuyModal &&
+      !!prevHand.distribution
+    );
+  }
+
+  return !!(
+    state === 'waiting' &&
+    myStack !== null && (myStack <= 0 || myStack < bb) &&
+    (state !== prevState || (myStack !== prevStack && prevStack > bb)) &&
```
Comment (troggy) [inline app/containers/Table/index.js]: what is this condition about? The logic is quite hard to follow

Comment (sunify) [reply app/containers/Table/index.js]: Check for changed hand state or changed stack. It's because we checking it in `componentWillReceiptProps`. Values can be updated at different moments of time.

But you right, it can be expressed in a more meaningful way. I'll try to refactor it.

Comment (sunify) [reply app/containers/Table/index.js]: I'll refactor this in different PR. It's pretty old condition and wasn't added in that PR

Comment (troggy) [reply app/containers/Table/index.js]: :+1:
